### PR TITLE
chore: Add cypress specs for partial import and export functionality

### LINKED ISSUE #30029: Task: Cypress test - partial import export
- [ ] Export flow
    - [ ] Export of a file and comparison of it with a pre-defined JSON file.
- [ ] Import flow
    - [ ] Success of partial import.
    - [ ]  Failure of partial import.

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description

This pull request adds partial import and export tests to the application. It includes the following changes:

- Adds exported file to be used in partial import export tests

- Add export and import options to EntityExplorer

- Adds spec to test export functionality.

- Add PartialImportExportLocators.ts, EntityCheckboxSelector.tsx, JSExportedOnly.json, and index.tsx

- Add test for partial export functionality for selected datasources

- Adds test to check all exported sections one by one

- Refactor partial export functionality tests

- Refactor export and import locators

- Add test for partial import functionality

#### PR fixes following issue(s)
Fixes #30029 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced partial export and import functionality for components such as JavaScript objects, queries, widgets, and data sources.
	- Enhanced Entity Explorer with new "Export" and "Import" actions.

- **Tests**
	- Added test cases for validating partial export and import features.
	- Updated test suite to include direct app level import with data sources.

- **Refactor**
	- Modified access level for the `uploadFile` property in the HomePage class to facilitate testing.

- **Style**
	- Updated UI components with test identifiers to improve testability.

- **Chores**
	- Updated the list of tests for running limited test scenarios.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/PartialImportExport/PartialImport_spec.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  sectionTitle: string,
+  elementsToCheck: string[],
+) {
+  cy.xpath(homePage._uploadFile).selectFile(
+    `cypress/fixtures/PartialImportExport/${fileName}`,
+    {
+      force: true,
+    },
+  );
+  agHelper.Sleep(3500);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/PartialImportExport/PartialImport_spec.ts:98]: Can this be replaced with some dynamic wait? LIke WaitUntilEleAppear or AssertingElementVisibility etc? Please do not add any static waits.

File: app/client/cypress/e2e/Regression/ClientSide/PartialImportExport/PartialImportExportLocators.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+export default {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/PartialImportExport/PartialImportExportLocators.ts:1]: Can this file please be added inside support/Pages?
If inside e2e, unless added into excludeSpecPattern in config file, this file will be considered spec & tried to be run by Cypress logic

Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/PartialImportExport/PartialImportExportLocators.ts]: Moved out. Thanks for this input.

File: app/client/cypress/e2e/Regression/ClientSide/PartialImportExport/PartialImport_spec.ts (hunk 1 above)
Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/PartialImportExport/PartialImport_spec.ts]: Damn... copy paste 😠 
Fixed now.

**Review by Aishwarya-U-R — APPROVED** [review]
Approving Cypress code changes.

File: app/client/cypress/e2e/Regression/ServerSide/AppLevelImport/AppImportwithDS_Spec.ts [hunk 3]
```diff
@@ -13,11 +13,6 @@ describe(
   "App level import with Datasource",
   { tags: ["@tag.ImportExport"] },
   () => {
-    before(() => {
-      homePage.CreateNewWorkspace("AppLevelImport", true);
-      homePage.CreateAppInWorkspace("AppLevelImport");
-    });
-
     //this tests coveres Applevel import for MySql, Gsheet is covered in Gsheet folder
```
Comment (rahulbarwal) [inline app/client/cypress/e2e/Regression/ServerSide/AppLevelImport/AppImportwithDS_Spec.ts:16]: @Aishwarya-U-R Can you please explain, why this is necessary? 
This spec fails without removing these lines.

**Review by jacquesikot — APPROVED** [review]

**Pull request merged**